src/utils/satelliteUtils.ts

```diff
@@ -19,12 +19,20 @@ export default class SatelliteUtilities {
         return null
     }
 
+    static async getSatellitesByRange(xStart: number, yStart: number, xEnd: number, yEnd: number): Promise<SatelliteDocument[]> {
```

Comment:
You can consider changing 4 numeric method arguments into 1 object argument, like:
```ts
type Range = {
xStart: number;
yStart: number;
xEnd: number;
yEnd: number;
};
...
static async getSatellitesByRange(range: Range)
```